### PR TITLE
Discard changes button is not taken into account

### PR DESCRIPTION
**Pull request by TawbaneAcc**
When designing a script we can modify an action (name, description and parameters)
image

If I modify a **parameter** (e.g name : "X") with a new value (e.g name : "ZZZZ") + do "**discard changes**"
image
image

The UI indicate that there is no new change (OK because -> Discard Changes)
image

But if I do "Save" and go back to the action, the value is **modified** (now name: "ZZZZ")
image

The issue is not encountered when we do the same steps with name and/or description, only for parameters.

**8416fd7**

### CODE REVIEW

File: src/views/design/ScriptDetail/index.tsx [hunk 1]
```diff
@@ -599,7 +599,7 @@ const ScriptDetail = withStyles(styles)(
             return (
                 <EditAction
                     onClose={() => this.setState({ editActionIndex: -1 })}
-                    action={this.getEditAction()}
+                    action={clone(this.getEditAction())}
```
Comment (BerrevoetsRobbe) [inline src/views/design/ScriptDetail/index.tsx:602]: is `this.getEditAction()` used anywhere else? Maybe this method should return the clone?

Comment (TawbaneAcc) [reply src/views/design/ScriptDetail/index.tsx]: yep, it is mainly used in EditAction component, and also in the variable editAction, but I think it has no impact on the code, I just checked it out, and nothing strange appear

File: src/views/design/ScriptDetail/index.tsx [hunk 2]
```diff
@@ -655,7 +655,7 @@ const ScriptDetail = withStyles(styles)(
             if (editActionIndex === -1) {
                 return null;
             }
-            return newScriptDetail && newScriptDetail.actions && newScriptDetail.actions[editActionIndex];
+            return clone(newScriptDetail && newScriptDetail.actions && newScriptDetail.actions[editActionIndex]);
```
Comment (BerrevoetsRobbe) [inline src/views/design/ScriptDetail/index.tsx:658]: shouldn't only the `newScriptDetail.actions[editActionIndex])` part be cloned?

**Review by BerrevoetsRobbe — APPROVED** [review]
approved

**Pull request merged**